### ISSUE TITLE
Calling up the collar menu via external script 

### ISSUE THREAD

**Issue by codeviolet** [OpenCollar/opencollar#541]
Hi, 

I don't think this will result in any code change/pull requests but thought this would be the best place to ask...

I'm designing a RLV script for creators to use in SL furniture. I'm the creator of a furniture script, and a fair number of users want to add 'RLV functionality' to their products.

I can't help but notice that I'm trying to re-implement many things that already exist in OpenCollar. For instance, dressing/undressing and RLV restrictions. So I'd rather send users directly to the OpenCollar menu than try to recreate all the RLV options within a single script in the furniture.

'menuto' seems to do what I want, but silently fails when the target avi is not on the owner's list (unless the collar wearer has set access to 'public'). 

For example, the following script is using the '[menuto](https://github.com/OpenCollar/opencollar/blob/main/LSL/OpenCollar%20-%20main.lsl#L399)' chat command:

``` javascript
string initials = "ar";
default{
    touch_start(integer total_number){
        llSay(1, initials+"menuto "+(string)llDetectedKey(0));
    }
}
```

So my question is:
Is there a best method to call up the collar menu for a non-owner via script (or perhaps a way for a scripted object to grant 'second' or 'temporary' owner status to an avatar)? In case it helps - the object sending the request will have already been granted RLV permissions i.e. Trust Object/Trust Owner.

Basically, I'm looking for guidance (or suggestions) about best practice for bringing up the OpenCollar menu via a furniture script. 

I could not find any instructions for 'menuto', aside from in the code - perhaps I am missing something really obvious!

Many thanks :)

**Comment by GarvinTwine** [comment]
For exactly what you want to do are RLV-Relays made for and not a collar.

http://wiki.secondlife.com/wiki/LSL_Protocol/Restrained_Love_Relay/Specification

**Comment by codeviolet** [comment]
Thanks, but that's not what I'm asking. 

OpenCollar is an RLV relay. Also the furniture has already been included as 'trusted object' by collar owner.

I'm asking for correct way for a script in the furniture to bring up collar menu, for a target avatar. Thus the target avatar can can then access the collar features (dressing/undressing and RLV restrictions) by pressing a button in the furniture menu.

``` javascript
<prefix>menuto <target_avatar_uuid>
```

This works, but only if the target avatar is already Owner/Second Owner.

I am seeking clarification about proper use of 'menuto', and also asking if a command exists to make request to add an avatar as a 'temporary owner' or 'second owner'.

The furniture sending the command has already been included as 'trusted object' by collar owner.

Kind Regards.

**Comment by GarvinTwine** [comment]
I answered you with advice and guidance and that is what you asked for.
OpenCollar is not a rlv-relay, in the current version it includes one, but that does not make OpenCollar a rlv-relay it is so much more than that.
And as you say the furniture has been included as 'trusted object' ... to the relay, which was just included once a long time ago to save attachment points and the need for a seperate relay. 
Collar commands are based on collar authorisation and so is menuto... and authorized can only be avatars and not objects.
I dont understand why you try to use the collar for things a relay was specially made for. 
So again my advice and recommendation:
Use rlv-relay protocol and your furniture can do anything rlv when the relay wearer grants (or has auto grants) permission.
cheers.

**Comment by codeviolet** [comment]
Agree that OpenCollar is more than a relay. That's why my question is asking if a best practice exists to bring up the collar menu for someone who is controlling the furniture, which is included as 'trusted object' by the relay. I was curious about giving the controlling avatar convenient access to collar menu (dressing/undressing & restrictions), to avoid unnecessary replication of these features in the furniture script.

Of course, furniture will make direct use of rlv relay protocol for all furniture related things. It's only the menus like dressing/undressing clothing/attachments/folders and comprehensive lists of restrictions which I feel the collar menu is better qualified to provide than a furniture script. Furniture is less likely to be updated by users than collar, when new rlv and sl features are added over time. And furniture has tighter script constraints because of land impact, etc. Yet people still demand convenient access to these type of long list of general rlv options from the furniture menu. So this is why.

Currently I have the furniture send 'menuto' command to the collar, when a button is pressed in the furniture menu. This does bring up collar menu directly from the furniture menu, but only if the target avatar is already on the collar's access list. So when the button is pressed, I let them know something like: _"Attempting to open the collar menu for [sitting avatar]. The avatar may first need to add you to their collar's access list..."_

I could not find any info about 'menuto' (only reference is in main.lsl) so I thought to ask before using it. Thus my question, specifically for case that the target avatar is not on collar's access list. Is there an approved method that a trusted object can prompt the collar wearer to add a new avatar temporarily to their collar access list? If not, then I feel it would be an excellent feature of OpenCollar if trusted objects were permitted to give certain menu access to avatars, for reasons and use case outlined above. 

Thanks very much for your reply.

Regards,
